tests/test_dataset.py

```diff
@@ -846,17 +846,72 @@ def test_features_transform_logic_graphdataset(self):
 
         dataset_test = GraphDataset(
             hdf5_path = hdf5_path,
-            target = 'binary',
             train = False,
             dataset_train = dataset_train,
-            features_transform = other_feature_transform
+            features_transform = other_feature_transform,
+            target = 'binary'
         )
         
         # features_transform setted in the testset should be ignored
         assert dataset_train.features_transform == dataset_test.features_transform
         assert dataset_train.means == dataset_test.means
         assert dataset_train.devs == dataset_test.devs
 
+    def test_inherit_info_training(self):
+        hdf5_path = "tests/data/hdf5/train.hdf5"
+        feature_transform = {'all': {'transform': None, 'standardize': True}}
+        
+        dataset_train = GraphDataset(
+            hdf5_path = hdf5_path,
+            node_features = ['bsa', 'hb_acceptors', 'hb_donors'],
+            edge_features = ['covalent', 'distance'],
+            features_transform = feature_transform,
+            target = 'binary',
+            target_transform = False,
+            task = 'classif',
+            classes = None
+        )
+        
+        dataset_valid = GraphDataset(
+            hdf5_path = hdf5_path,
+            train = False,
+            dataset_train = dataset_train,
+        )
+        
+        # node_features, edge_features, features_dict, feature_transform, target, target_transform, task, and classes 
+        # in the valid should be inherited from the train
+        
+        assert dataset_train.node_features == dataset_valid.node_features
+        assert dataset_train.edge_features == dataset_valid.edge_features
+        assert dataset_train.features_dict == dataset_valid.features_dict
+        assert dataset_train.features_transform == dataset_valid.features_transform
+        assert dataset_train.target == dataset_valid.target
+        assert dataset_train.target_transform == dataset_valid.target_transform
+        assert dataset_train.task == dataset_valid.task
+        
+        dataset_test = GraphDataset(
```

Comment:
```suggestion
        dataset_valid = GraphDataset(
```
Let's stick to the same name